website/blog/2024-05-22-whats-next.md

```diff
@@ -0,0 +1,75 @@
+
+Given MACI's open source nature, it's common for our core team to develop new features or to fix issues based on community feedback. However, it's been less common for external contributors to make significant changes to the core protocol.
+
+Well, this soon will be a reality thanks to a MACI improvement proposal sent by the [3327](https://3327.io/) team. 3327 is collective of 10+ people working on improving blockchain technologies, with a focus on research and engineering. Their development team will be working to implement significant changes to the MACI's codebase.
+
+This proposal can be divided into two parts:
+
+1. bring unconditional privacy to MACI's voters
+2. optimise inefficient merkle tree structure holding messages, by replacing it with a hash chain
+
+## 1) Provide unconditional voter privacy
+
+Currently with MACI, if a voter performs a [key change](/docs/core-concepts/key-change), the voter's new key would not be anonymous to the coordinator. The coordinator could collude with a bad actor to inform the latter of the key change, as the coordinator would have access to all decrypted messages.
+
+The key focus of this improvement is to allow users to be completely anonymous by removing the link between the original signup key and the key used for voting. How would this work? Well, users sign up to vote via the [MACI contract](/docs/developers-references/smart-contracts/MACI), and depending on the gatekeeper in use, they'd have to prove that they've passed the entry condition. Now, given knowledge of this key, they can signup with a new key to polls deployed by this same MACI contract.
+
+Thus, voters can prove anonymously that they know the preimage of a [`StateLeaf`](/docs/developers-references/typescript-code/typedoc/domainobjs/classes/StateLeaf), by passing this information to a zk-SNARK circuit, and validating this proof within the poll contract when joining with the new key. You might be thinking that everyone knows the preimage of a state leaf, as it's public information that can be taken from the contracts' logs. However, the circuit will not accept the public key directly but would instead take the private key and use it to generate the public key. This way, only users with knowledge of a specific private key can generate a valid inclusion proof.
+
+Now after signing up to the Poll with this new key, there will not be any link to the original key, and users will effectively be anonymous. Of course users should ensure that they are using different wallets where possible.
+
+Finally, with the use of a nullifier, it will not be possible for the same original key to be used to signup more than once for each new poll.
+
+Are there any drawbacks? Well, yes. There will be an extra step for users to register to individual Polls. We aim to offset this cost and additional step soon either with gasless transactions or by moving some logic off-chain.
```

Comment:
This can be done atomically on sign up

Reply:
You mean signup to MACI's contract then signup to a new poll with a different key?

Reply:
yes exactly so in the old flow it would have been: [signup + deactivate message from inclusion proof] on core contract, then [rerandomize key + vote] on poll.